### DISCUSSION TITLE
Send an embed with multiple images

### DISCUSSION THREAD

**Discussion by yannic-md** [Rapptz/discord.py#9045] [General] [answered]
https://www.reddit.com/r/discordapp/comments/raz4kl/finally_a_way_to_display_multiple_images_in_an/

How is something like that possible in discord.py? I already tried to use the advice from the reddit guy and have this code:
```
if message.attachments:
                for entry in message.attachments:
                    if "image" in str(entry.content_type):
                        image_file = True
                        if not embed.image:
                            embed.set_image(url=str(entry.url))
                        else:
                            new_embed = discord.Embed()
                            new_embed.set_image(url=str(entry.url))
                            embed_list.append(new_embed)
                await message.reply(embeds=[embed_list])
```

But that just sends multiple embeds with one picture each.

**Comment by LostLuma** [comment, marked as answer]
Hello, as the Reddit posts states, all of your embeds must have the same URL property set in addition to the image URL:
`The key here is to use the same URL for all of the 4 embeds! I used my bot's website as the URL for the "url" property.`

With discord.py you can pass the `url` keyword parameter to the [`Embed` constructor](https://discordpy.readthedocs.io/en/latest/api.html#discord.Embed) to achieve this effect:
```py
embed = discord.Embed(url='https://lostluma.dev').set_image(url='https://files.lostluma.dev/9X75qL.gif')
```

The URL is not visible in the embed unless you also pass the `title` parameter, so you can use any dummy URL you'd like to.

**Reply by yannic-md** [reply]
I think that **this** should really be documented.
I know so much people that doesn't know this is possible.

**Reply by Rapptz** [reply]
It's an undocumented client quirk, not even Discord documents it because it's a hack their unfurler uses for Twitter links.

**Reply by Radon333** [reply]
Writing code in Elixir here using Nostrum. 

I have set the url parameter same but I get two different embeds for the images. Any idea how this can be resolved and i can get multiple images in one embed itself?

```
    embed1 =
      %Nostrum.Struct.Embed{}
      |> put_url("https://example.org/")
      |> put_image("https://cdn.discordapp.com/ephemeral-attachments/1099587079010254878/1109038810559430676/d98060a4-198f-4d7a-9253-36f84cc8c9a4-2.png")

    embed2 =
      %Nostrum.Struct.Embed{}
      |> put_url("https://example.com/")
      |> put_image("https://cdn.discordapp.com/attachments/1099587079010254878/1108966211565391982/test.png")

    Api.create_message(msg.channel_id, embeds: [embed1, embed2], components: [action_row])
  end
end
```

**Reply by tailoric** [reply]
The urls are not the same tho
One is a .org the other is a .com

**Reply by Radon333** [reply]
sorry didn't notice

**Comment by huths0lo** [comment]
I wanted to post this for future travellers that may find this.  I do use python, and do use the Pycord library.  But I have allot of very custom code that just needs to work outside of a library.  Sending messages in to discord is not difficult to do without a library, as long as you know how to structure the payload.  The problem with every single bit of documentation is it explains the commands to build what you want with a library, but doesnt explain at all what the payload looks like; because the library builds your payload and sends it for you.

I've been trying to make multi image payloads for a year now, and had given up; leaving it to just be a column of images.  But I again had a real need for this.  Its this answer above that finally gave me the missing clue.  And its not a surprise its taken this long, since the answer above only arrived November 2022, and I havent tried making one for some time now.

Here is a working multi image, single row payload.  Thank you @LostLuma for providing the missing link.

`{
	"content": "P2P Notice | New Listing #3.\\n\\nTrading: \\u20b350.0 and 1 x Cardano Budz H2 #965 and 1 x Cardano Budz H1 #0068 in exchange for 1000 rupees.",
	"embeds": [{
		"color": "16776960",
		"image": {
			"url": "https://ipfs.io/ipfs/QmYsvXSq5k1Xk5rQKHo9A5ZUiDJS4ayhLcGfhCKvGKAQrj"
		},
		"url": "https://themorphium.io/match_maker/listing/3"
	}, {
		"color": "16776960",
		"image": {
			"url": "https://ipfs.io/ipfs/QmZsGz4oLsBYcyDeqnGftjQ5Z7zTgKah5qzxFd2YT8m8Wv"
		},
		"url": "https://themorphium.io/match_maker/listing/3"
	}]
}`
